### PR TITLE
feat(sdui-runtime): TabsFooter renderer — renders tab nodes, dispatches on_click, active-index styling

### PR DESCRIPTION
**Pull request by mridulgupta-oi**
## Summary

Brings the `TabsFooter` snippet (`creator.snippet.tabs_footer`) up to legacy parity with `TabsFooterSnippetType1` from the legacy app:

- Renders each tab Node in an equal-width slot (`flex: 1` per tab).
- Adds the top hairline border + soft top-edge box shadow from the legacy snippet so the footer reads as a pinned bottom navigation.
- Reads `data.active_index` and overrides each child tab Node's `data.active` flag — the inner `Tab` renderer (`creator.ui_component.tab`) already paints the active state via `data.active`, so the active tab gets the primary-color tint automatically.
- `on_click` is dispatched per-tab via the existing Tab → `SduiNode` → `Clickable` chain — no extra wrapping layer in `TabsFooter`.
- Active-index override is non-mutating; tabs without `active_index` keep their producer-supplied `data.active`.

The renderer was already registered for `creator.snippet.tabs_footer` — no registry changes needed.

## Why

The home page in Brief 264 ships a `tabs_footer` Node inside the new `PageFeed` `data.footer` slot (see [#177](https://github.com/One-Impression/amplify-design-system/pull/177)). The existing `TabsFooter` renderer was a placeholder that didn't:

- give each tab equal width
- support active-index selection
- apply the legacy top-edge shadow

This PR closes those gaps so home renders correctly without further patches.

## Notes

- `data.active_index` schema field already exists in `@one-impression/sdk-native-sdui` `TabsFooterSchema` (`active_index: z.number().int().optional()`), so no schema changes are required.
- The `applyActiveIndex` helper is factored out as a pure function so it can be unit-tested without a React Native runtime.

## Test plan

- [x] `npm run build -w packages/sdui-runtime` — clean tsup + tsc build
- [x] `npm test -w packages/sdui-runtime` — 7 new tests in `applyActiveIndex.test.ts` pass (pre-existing `branch.test.ts` / `compound.test.ts` / `bff-call.test.ts` / `set-local.test.ts` / `node-registry.test.ts` peer-dep failures unchanged — known issue)
- [ ] Manual: render a TabsFooter with 3 tabs, verify each renders in an equal-width slot
- [ ] Manual: set `data.active_index = 1`, verify middle tab gets primary tint and others reset
- [ ] Manual: tap an inactive tab, verify the tab's own `on_click` action dispatches

### CODE REVIEW

File: packages/sdui-runtime/src/snippets/TabsFooter/TabsFooter.renderer.tsx [hunk 1]
```diff
@@ -1,10 +1,56 @@
...
+  /**
+   * Top-level container — pinned-bottom row of tabs with a hairline top
+   * border and a soft top-edge shadow (matches legacy
+   * `TabsFooterSnippetType1.styles.ts` box shadow).
+   */
+  container: {
+    flexDirection: "row",
+    paddingHorizontal: 16,
+    paddingVertical: 12,
+    backgroundColor: "#FFFFFF",
```
Comment (zenith-oi) [inline packages/sdui-runtime/src/snippets/TabsFooter/TabsFooter.renderer.tsx:19]: **FLAG** — [no-hardcoded-colors (CLAUDE.md rule 1) / no-hardcoded-business-values] The StyleSheet inlines four design-system values that should come from tokens: `backgroundColor: '#FFFFFF'` (L19), `borderTopColor: '#E0E0E0'` (L21), `shadowColor: 'rgba(0, 0, 0, 0.15)'` (L22), and the padding constants `16` / `12` (L17-18). CLAUDE.md rule 1 explicitly states 'No hardcoded colors in UI components — use CSS variables only', and the monorepo ships `packages/tokens-creator/` with SDUI-mapped tokens for exactly this purpose. Hardcoding these values means a theme change (e.g. dark-mode background, brand-specific border color) requires a code change here rather than a token update. Fix: import the relevant token values from `@one-impression/tokens-creator` (or the appropriate token package) and reference them in the StyleSheet. Also applies to `paddingHorizontal: 16` and `paddingVertical: 12` — these should be spacing tokens.

_Cited:_ `CLAUDE.md#rules — 'No hardcoded colors in UI components — use CSS variables only'`

**Review by zenith-oi — DISMISSED** [review]
## Zenith Code Review

**Clean, well-scoped renderer upgrade. The `applyActiveIndex` helper is correctly factored as a pure function with solid test coverage. One minor concern: hardcoded color literals (`#FFFFFF`, `#E0E0E0`, `rgba(0,0,0,0.15)`) and spacing values (`16`, `12`) in the StyleSheet are inlined rather than drawn from the design-system token packages that exist in this monorepo — worth flagging given the repo's own no-hardcoded-colors rule in CLAUDE.md.**

### What this PR is doing
**Intent:** Bring the TabsFooter SDUI snippet renderer to legacy parity by adding equal-width tab slots, a top-border/shadow matching the legacy app, and active-index-driven `data.active` override via a pure, tested `applyActiveIndex` helper.
**Delivered:** The diff fully delivers the stated intent. The renderer now uses RN `View` + `StyleSheet` for the container and per-tab slots (equal-width `flex: 1`), applies the legacy top-border and shadow, and delegates active-state management to `applyActiveIndex`. The helper is pure, non-mutating, and covered by 7 unit tests including edge cases (undefined index, out-of-range, null data, empty array, on_click preservation). The test runner glob is wired in `package.json`. No registry changes were needed (confirmed in PR body). No acceptance items from a linked task page are present, but the PR body's own checklist maps cleanly to the diff.

**1 FLAG** — advisory, not merge-blocking. See inline comments.

**Positive:**
- Extracting `applyActiveIndex` as a pure function with no React/RN dependency is exactly the right call — it makes the logic independently testable without a renderer runtime.
- The test suite is thorough: covers the undefined-passthrough fast path (identity return), mutation guard (JSON.stringify before/after), out-of-range index, null data defensive pass-through, empty array, and on_click preservation. Seven tests for a 34-line helper is good coverage discipline.
- The changeset entry is detailed and accurate — it documents the legacy reference, the active-index contract, and the non-mutating guarantee, which is exactly what downstream consumers need to understand the snippet's behaviour.

**Read against:**
- `PR body — stated goals: equal-width slots, top-border/shadow, active_index override, non-mutating, no registry changes`
- `PR body — test plan: build clean, 7 new tests pass`

---
*Generated by [Zenith AI CTO](https://github.com/one-impression/zenith-coding-agent)*

**Review by zenith-oi — DISMISSED** [review]
## Zenith Code Review

**Clean, well-scoped renderer upgrade with a properly isolated pure helper and solid test coverage. No issues worth blocking or flagging — the hardcoded color/spacing values in the stylesheet are UI constants that match a specific legacy visual spec, not business-logic literals, and the `no-hardcoded-business-values` pattern explicitly exempts this kind of UI styling from its scope.**

### What this PR is doing
**Intent:** Bring the `TabsFooter` SDUI renderer to legacy parity by adding equal-width tab slots, a top-border/shadow matching the legacy `TabsFooterSnippetType1`, and `active_index`-driven active-state override via a pure, non-mutating `applyActiveIndex` helper.
**Delivered:** The diff fully delivers the stated intent. The renderer now uses `StyleSheet`/`View` with `flex: 1` per tab slot, applies the hairline border and shadow, reads `v.active_index` via `applyActiveIndex`, and the helper is factored out as a pure function with 7 unit tests covering the happy path, mutation safety, edge cases (empty array, out-of-range index, null data), and on_click preservation. The test script in `package.json` is updated to include the new test file. No registry changes were needed (confirmed in PR body). All stated acceptance criteria are met.

No issues found.

**Positive:**
- Factoring `applyActiveIndex` into a pure, side-effect-free function that can be tested without a React Native runtime is exactly the right call — it makes the active-index logic independently verifiable and reusable.
- The non-mutation guarantee is explicitly tested (`does not mutate input items`) and enforced in the implementation via spread — good defensive practice for SDUI node trees that may be shared across renders.
- Edge-case coverage is thorough: empty array, out-of-range index, null `data`, and on_click preservation are all explicitly tested, not just the happy path.
- The changeset description is unusually clear — it explains the legacy reference, the design decision (declarative `active_index` vs `useNavigationState`), and the dispatch chain, which will be valuable context for future maintainers.

---
*Generated by [Zenith AI CTO](https://github.com/one-impression/zenith-coding-agent)*

File: packages/sdui-runtime/src/snippets/TabsFooter/TabsFooter.renderer.tsx (hunk 1 above)
Comment (zenith-oi) [inline packages/sdui-runtime/src/snippets/TabsFooter/TabsFooter.renderer.tsx:19]: **FLAG** — [no-hardcoded-colors (CLAUDE.md rule 1)] The `container` style inlines three color literals — `#FFFFFF` (background), `#E0E0E0` (border), and `rgba(0, 0, 0, 0.15)` (shadow) — and two spacing values (`paddingHorizontal: 16`, `paddingVertical: 12`). CLAUDE.md rule 1 states: "No hardcoded colors in UI components — use CSS variables only." For React Native the equivalent is consuming tokens from `@one-impression/tokens-creator` (or the relevant token package) rather than inlining raw hex/rgba values. The spacing values should similarly come from the foundation spacing scale. Fix: import the relevant token constants (e.g. `colors.surface.default`, `colors.border.subtle`, `colors.shadow.overlay`, `spacing.md`) and reference them here. If the token package doesn't yet export these values, add them there first so all three products benefit — that's the repo's stated composition rule.

_Cited:_ `CLAUDE.md#rules — 'No hardcoded colors in UI components — use CSS variables only'`

**Review by zenith-oi — DISMISSED** [review]
## Zenith Code Review

**Clean, well-scoped renderer upgrade. The `applyActiveIndex` extraction is the right call — pure function, fully tested, non-mutating. One minor concern worth flagging: hardcoded color literals (`#FFFFFF`, `#E0E0E0`, `rgba(0,0,0,0.15)`) and spacing values (`16`, `12`) in the `StyleSheet` are inlined rather than drawn from the design-token packages this repo exists to maintain, which is the canonical anti-pattern for this codebase per CLAUDE.md rule 1.**

### What this PR is doing
**Intent:** Bring the `TabsFooter` SDUI renderer to legacy parity by adding equal-width tab slots, a top-border/shadow matching the legacy `TabsFooterSnippetType1`, and `active_index`-driven active-state override via a new pure `applyActiveIndex` helper — inferred from diff + body.
**Delivered:** The diff fully delivers the stated intent. The renderer now uses `StyleSheet`/`View` with `flex: 1` per tab slot, applies the hairline border and shadow, and delegates active-index logic to the extracted `applyActiveIndex` helper. The helper is non-mutating, handles undefined/out-of-range/null-data defensively, and is covered by 7 unit tests. The test command in `package.json` is updated to include the new test file. No registry changes were needed (confirmed in PR body). No acceptance items from a linked task page are present; the PR body's checklist items are all addressed.

**1 FLAG** — advisory, not merge-blocking. See inline comments.

**Positive:**
- Extracting `applyActiveIndex` as a pure, framework-free function is exactly the right design — it makes the active-index logic independently testable without a React Native runtime and keeps the renderer thin.
- The test suite is thorough: covers the identity path (undefined activeIndex), mutation-safety, on_click preservation, defensive null-data passthrough, empty array, and out-of-range index. Edge cases that often get missed are all present.
- The changeset entry is detailed and accurate — it documents the legacy reference, the non-mutating contract, and the producer opt-out behaviour, which is exactly the level of context future engineers need.
- Removing the `Box`/`Stack` abstraction in favour of direct `StyleSheet.create` + `View` is the right call for a pinned navigation bar where pixel-precise shadow/border control matters and the higher-level primitives would fight the layout.

---
*Generated by [Zenith AI CTO](https://github.com/one-impression/zenith-coding-agent)*

File: packages/sdui-runtime/src/snippets/TabsFooter/TabsFooter.renderer.tsx (hunk 1 above)
Comment (zenith-oi) [inline packages/sdui-runtime/src/snippets/TabsFooter/TabsFooter.renderer.tsx:19]: **FLAG** — [no-hardcoded-colors (CLAUDE.md Rule 1)] Three hardcoded color literals appear in the `StyleSheet`: `backgroundColor: "#FFFFFF"` (L19), `borderTopColor: "#E0E0E0"` (L21), and `shadowColor: "rgba(0, 0, 0, 0.15)"` (L22). CLAUDE.md Rule 1 states: "No hardcoded colors in UI components — use CSS variables only." In a React Native context this means design tokens from `@one-impression/tokens-creator` (or the equivalent RN token surface) rather than inline hex/rgba values. The legacy snippet these values are copied from is not a justification for carrying the anti-pattern forward into the new renderer. Fix: resolve the equivalent semantic tokens (e.g. `colors.surface.default`, `colors.border.subtle`, `shadows.bottom-nav`) and reference them here. If the token doesn't exist yet, add it to `tokens-creator` in a companion PR.

_Cited:_ `CLAUDE.md Rule 1 — "No hardcoded colors in UI components — use CSS variables only"`

Comment (zenith-oi) [inline packages/sdui-runtime/src/snippets/TabsFooter/TabsFooter.renderer.tsx:19]: **FLAG** — [no-hardcoded-colors (CLAUDE.md Rule 1)] Four design-system values are still hardcoded in the StyleSheet: `backgroundColor: '#FFFFFF'` (L19), `borderTopColor: '#E0E0E0'` (L21), `shadowColor: 'rgba(0, 0, 0, 0.15)'` (L22), and the padding literals `paddingHorizontal: 16` / `paddingVertical: 12` (L17–18). CLAUDE.md Rule 1 is explicit: "No hardcoded colors in UI components — use CSS variables only." For React Native the equivalent is consuming tokens from the design-system token packages (e.g. `tokens-creator` or `tokens-foundation`) rather than inlining raw hex/rgba values and magic spacing numbers. These should be replaced with token imports so Pixel can detect drift and the values stay in sync across products.

_Cited:_ `CLAUDE.md#rules-1-no-hardcoded-colors`

**Review by zenith-oi — APPROVED** [review]
## Zenith Code Review

**Clean, well-scoped PR that brings TabsFooter to legacy parity. The only outstanding issue is the carry-over from the previous review round: four design-system color/spacing values are hardcoded in the StyleSheet rather than sourced from tokens, which violates CLAUDE.md Rule 1. Everything else — the pure helper, immutability, test coverage, changeset, and package.json update — is solid.**

### What this PR is doing
**Intent:** Bring the TabsFooter SDUI snippet renderer to legacy parity by adding equal-width tab slots, a top-border/shadow matching the legacy app, and active-index-driven tab highlighting via a non-mutating `applyActiveIndex` helper.
**Delivered:** The diff fully delivers the stated intent. The renderer now uses `StyleSheet`-based layout with `flex: 1` per tab slot, a hairline top border and shadow matching the legacy `TabsFooterSnippetType1`, and the `applyActiveIndex` pure function correctly overrides `data.active` per tab without mutating inputs. Seven unit tests cover the helper's edge cases (undefined index, out-of-range, empty array, bogus data, on_click preservation, immutability). The test script in `package.json` is updated to include the new test file. No linked task page or brief with numbered acceptance items was present, so no acceptance-item gaps to report.

> ℹ️ 2 previous Zenith reviews dismissed. Inline comments from those reviews are superseded — only comments from **this review** are current.

**1 FLAG** — advisory, not merge-blocking. See inline comments.

**Positive:**
- `applyActiveIndex` is correctly factored as a pure, non-mutating function — the early-return on `undefined` preserves producer-supplied `data.active` values without any extra branching in the renderer.
- Test coverage is thorough: undefined index, out-of-range index, empty array, bogus `data: null` node, on_click preservation, and the immutability invariant are all explicitly asserted.
- Dropping the `Box`/`Stack` abstraction in favour of direct `View` + `StyleSheet` is the right call for a performance-sensitive bottom-nav component rendered on every frame of the home page.
- The changeset entry is detailed and accurately describes the behaviour change, the legacy reference, and the producer contract — good for downstream consumers tracking the changelog.

**Read against:**
- `CLAUDE.md#rules-1-no-hardcoded-colors`
- `patterns/no-hardcoded-business-values`

---
*Generated by [Zenith AI CTO](https://github.com/one-impression/zenith-coding-agent)*

**Review by zenith-oi — APPROVED** [review]
## Zenith Code Review

**Clean, well-scoped renderer upgrade. The `applyActiveIndex` helper is correctly pure and non-mutating, the StyleSheet values match the stated legacy reference, and the test coverage is thorough. One minor concern worth flagging: hardcoded color literals (`#FFFFFF`, `#E0E0E0`, `rgba(0,0,0,0.15)`) in the renderer rather than design tokens — this repo's CLAUDE.md Rule 1 explicitly forbids hardcoded colors in UI components.**

### What this PR is doing
**Intent:** Bring the `TabsFooter` SDUI renderer to legacy parity by adding equal-width tab slots, a top-border/shadow matching the legacy `TabsFooterSnippetType1`, and `active_index`-driven active-state override via a pure, non-mutating `applyActiveIndex` helper.
**Delivered:** The diff fully delivers the stated intent. The renderer now uses `StyleSheet`-based `View` layout with `flex: 1` per tab slot, hairline border + shadow matching the legacy snippet, and the `applyActiveIndex` helper correctly overrides `data.active` per tab without mutating inputs. The helper is factored out as a pure function with 7 unit tests covering the happy path, immutability, edge cases (empty array, out-of-range index, null data, undefined activeIndex), and on_click preservation. The test suite is wired into `package.json`. No registry changes were needed (confirmed in PR body). All stated acceptance criteria are met.

**1 FLAG** — advisory, not merge-blocking. See inline comments.

**Positive:**
- The `applyActiveIndex` helper is a textbook pure function: no mutation, early-return for the undefined case, defensive pass-through for nodes with non-object data, and a generic `<T extends Node>` signature that preserves the concrete type through the map.
- Test coverage is genuinely thorough — 7 cases cover the happy path, immutability invariant, empty array, out-of-range index, null data guard, and on_click preservation. The immutability test using `JSON.stringify` before/after is a clean, dependency-free approach.
- Removing the `Box`/`Stack` abstraction layer in favour of direct `View` + `StyleSheet` is the right call for a pinned navigation bar where the exact shadow/border values must match legacy pixel-for-pixel.
- The changeset entry is detailed and accurate — it explains the design rationale, the legacy reference, and the producer contract clearly.

---
*Generated by [Zenith AI CTO](https://github.com/one-impression/zenith-coding-agent)*
